src/wstool/multiproject_cmd.py

```diff
@@ -274,6 +276,32 @@ def do_work(self):
     return outputs
 
 
+def cmd_foreach(config, command, shell=False):
+    """Run command in all SCM entries in config, relative to path"""
+
+    class ForeachRetriever(object):
+        def __init__(self, element, command, shell):
+            self.element = element
+            self.command = command
+            self.shell = shell
+
+        def do_work(self):
+            command = self.command
+            if not self.shell:
+                command = shlex.split(command)
+            po = Popen(command, cwd=self.element.path, shell=self.shell,
```

Comment:
Also, run_shell_command's output has unexpected new lines like below:

```
$ ../hoge.py
* add-foreach-command

  another-wstool-completion

  master

  wstool-foreach

------
(0, u'* add-foreach-command\n\n  another-wstool-completion\n\n  master\n\n  wstool-foreach', None)
```

hoge.py is
```
#!/usr/bin/env python
# -*- coding: utf-8 -*-

from vcstools.common import run_shell_command

cmd = 'git branch'
o = run_shell_command(cmd, shell=True, show_stdout=True, verbose=False, no_warn=True, no_filter=False)

print('------')
print(o)
```